src/time_integration/paired_explicit_runge_kutta/methods_PERK3.jl

```diff
@@ -0,0 +1,366 @@
+    u::uType
+    du::uType
+    u_tmp::uType
+    t::RealT
+    tdir::RealT
+    dt::RealT # current time step
+    dtcache::RealT # manually set time step
+    iter::Int # current number of time steps (iteration)
+    p::Params # will be the semidiscretization from Trixi
+    sol::Sol # faked
+    f::F
+    alg::Alg # This is our own class written above; Abbreviation for ALGorithm
+    opts::PairedExplicitRKOptions
+    finalstep::Bool # added for convenience
+    dtchangeable::Bool
+    force_stepfail::Bool
+    # PairedExplicitRK stages:
+    k1::uType
+    k_higher::uType
+end
+
+function init(ode::ODEProblem, alg::PairedExplicitRK3;
+              dt, callback = nothing, kwargs...)
+    u0 = copy(ode.u0)
+    du = zero(u0)
+    u_tmp = zero(u0)
+
+    # PairedExplicitRK stages
+    k1 = zero(u0)
+    k_higher = zero(u0)
+
+    t0 = first(ode.tspan)
+    tdir = sign(ode.tspan[end] - ode.tspan[1])
+    iter = 0
+
+    integrator = PairedExplicitRK3Integrator(u0, du, u_tmp, t0, tdir, dt, dt, iter,
+                                             ode.p,
+                                             (prob = ode,), ode.f, alg,
+                                             PairedExplicitRKOptions(callback,
+                                                                     ode.tspan;
+                                                                     kwargs...),
+                                             false, true, false,
+                                             k1, k_higher)
+
+    # initialize callbacks
+    if callback isa CallbackSet
+        for cb in callback.continuous_callbacks
+            error("unsupported")
+        end
+        for cb in callback.discrete_callbacks
+            cb.initialize(cb, integrator.u, integrator.t, integrator)
+        end
+    elseif !isnothing(callback)
+        error("unsupported")
+    end
+
+    return integrator
+end
+
+# Fakes `solve`: https://diffeq.sciml.ai/v6.8/basics/overview/#Solving-the-Problems-1
+function solve(ode::ODEProblem, alg::PairedExplicitRK3;
+               dt, callback = nothing, kwargs...)
+    integrator = init(ode, alg, dt = dt, callback = callback; kwargs...)
+
+    # Start actual solve
+    solve!(integrator)
+end
+
+function solve!(integrator::PairedExplicitRK3Integrator)
+    @unpack prob = integrator.sol
+
+    integrator.finalstep = false
+
+    @trixi_timeit timer() "main loop" while !integrator.finalstep
+        step!(integrator)
+    end # "main loop" timer
+
+    return TimeIntegratorSolution((first(prob.tspan), integrator.t),
+                                  (prob.u0, integrator.u),
+                                  integrator.sol.prob)
+end
+
+function step!(integrator::PairedExplicitRK3Integrator)
+    @unpack prob = integrator.sol
+    @unpack alg = integrator
+    t_end = last(prob.tspan)
+    callbacks = integrator.opts.callback
+
+    @assert !integrator.finalstep
+    if isnan(integrator.dt)
+        error("time step size `dt` is NaN")
```

Comment:
Is this the same error message as in OrdinaryDifEq.jl for consistency?

Reply:
I don't think so. But as far as I understand OrdinaryDiffEq.jl handles these scenarios in step! rather differently than how we do it. As mentioned below, if we want to prioritize consistency, IMHO, it might be a better idea to start another issue regarding this.

Reply:
Ok, we can think of this in another issue.